src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateWorkitem.sql

```diff
@@ -0,0 +1,211 @@
+﻿/*************************************************************
+    Stored procedure to Update a workitem.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     UpdateWorkitem
+--
+-- DESCRIPTION
+--     Update a UPS-RS Workitem.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The system identifier of the data partition.
+--     @workitemUid
+--         * The workitem UID.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+--     @status
+--         * Status of the workitem, Either 1(ReadWrite) or 2(Read)
+-- RETURN VALUE
+--     WorkitemKey
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.UpdateWorkitem
+    @partitionKey                   INT,
+    @workitemUid                    VARCHAR(64),
+
+    @stringExtendedQueryTags        dbo.InsertStringExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags      dbo.InsertDateTimeExtendedQueryTagTableType_2 READONLY,
+    @personNameExtendedQueryTags    dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY,
+
+    @status                         TINYINT
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @workitemKey BIGINT
+    DECLARE @newWatermark BIGINT
+    DECLARE @workitemResourceType TINYINT = 1
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+
+    -- Get the workitem key
+    SELECT @workitemKey = WorkitemKey
+    FROM dbo.Workitem
+    WHERE
+        PartitionKey = @partitionKey
+        AND WorkitemUid = @workitemUid
+
+    IF @@ROWCOUNT = 0
+        THROW 50413, 'Workitem does not exist', 1;
+
+    SET @newWatermark = NEXT VALUE FOR dbo.WatermarkSequence
+
+    -- String Key tags
+    IF EXISTS (SELECT 1 FROM @stringExtendedQueryTags)
+    BEGIN
+
+        WITH InputCTE AS (
```

Comment:
Execution Plan for the CTE Query

[UpdateWorkitem_InputCTE_Query_ExecutionPlan.zip](https://github.com/microsoft/dicom-server/files/8003111/UpdateWorkitem_InputCTE_Query_ExecutionPlan.zip)